var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -68,27 +65,34 @@ class Hiop(CMakePackage, CudaPackage):
     depends_on('magma@2.6.1:', when='@0.4.6:+cuda')
 
     depends_on('raja+openmp', when='+raja')
+    depends_on('raja@0.14.0:', when='@0.5.0:+raja')
     depends_on('raja+cuda', when='+raja+cuda')
-    depends_on('umpire+cuda', when='+raja+cuda')
+    depends_on('umpire', when='+raja')
+    depends_on('umpire+cuda~shared', when='+raja+cuda')
 
     depends_on('suite-sparse', when='+kron')
 
     depends_on('coinhsl+blas', when='+sparse')
     depends_on('metis', when='+sparse')
 
+    conflicts(
+        '+shared',
+        when='+cuda+raja',
+        msg='umpire+cuda exports device code and requires static libs',
+    )
+
     flag_handler = build_system_flags
 
     def cmake_args(self):
         args = []
         spec = self.spec
 
-        lapack_blas_libs = (
-            spec['lapack'].libs + spec['blas'].libs).joined(';')
-        args.extend([
-            '-DLAPACK_FOUND=TRUE',
-            '-DLAPACK_LIBRARIES={0}'.format(lapack_blas_libs)
-        ])
+        lapack_blas_libs = (spec['lapack'].libs + spec['blas'].libs).joined(';')
+        args.extend(
+            ['-DLAPACK_FOUND=TRUE', '-DLAPACK_LIBRARIES={0}'.format(lapack_blas_libs)]
+        )
```

Comment:
Prefer using the cmake package helper functions, which convert bools and lists to their appropriate CMake string representation:
```suggestion
        args.extend([
            self.define('LAPACK_FOUND', True),
            self.define('LAPACK_LIBRARIES',
                        spec['lapack'].libs + spec['blas'].libs)
        ])
```